crates/katana/pool/src/pool.rs

```diff
@@ -110,12 +136,14 @@ where
                         let tx = PendingTx::new(id, tx, priority);
 
                         // insert the tx in the pool
-                        self.inner.transactions.write().insert(tx);
-                        self.notify_listener(hash);
+                        self.inner.transactions.write().insert(tx.clone());
+                        self.notify(tx);
 
                         Ok(hash)
                     }
 
+                    // TODO: create a small cache for rejected transactions to respect the rpc spec
+                    // `getTransactionStatus`
```

Comment:
Don't we store the rejected transactions anyway? Or the cache is just for improvement?

Reply:
no we dont store them yet,

https://github.com/dojoengine/dojo/blob/942f6f14c2eede7b4d060015e7264e98479a7663/crates/katana/pool/src/pool.rs#L147-L150

they are dropped instantly here

Reply:
meaning, the `getTransactionStatus` endpoint can never return a tx status that is `REJECTED` 

https://github.com/starkware-libs/starknet-specs/blob/6d88b7399f56260ece3821c71f9ce53ec55f830b/api/starknet_api_openrpc.json#L2976C7-L2981C9